subgraph/src/KlerosCore.ts

```diff
@@ -126,10 +126,21 @@ export function handleNewPeriod(event: NewPeriod): void {
   }
 
   dispute.period = newPeriod;
-  dispute.lastPeriodChange = event.block.timestamp;
+  dispute.lastPeriodChangeTs = event.block.timestamp;
+  dispute.lastPeriodChangeBlockNumber = event.block.number;
+  let counter = PeriodIndexCounter.load(newPeriod);
+  if (!counter) {
+    counter = new PeriodIndexCounter(newPeriod);
+    counter.counter = BigInt.fromI32(0);
+  }
+  dispute.periodNotificationIndex = counter.counter;
+  counter.counter = counter.counter.plus(ONE);
+  counter.save();
   dispute.lastPeriodChangeBlockNumber = event.block.number;
   if (newPeriod !== "execution") {
-    dispute.periodDeadline = event.block.timestamp.plus(court.timesPerPeriod[event.params._period]);
+    const contract = KlerosCore.bind(event.address);
```

Comment:
If possible, I would prefer to keep the use of the subgraph's `timesPerPeriod` from the court entity, rather than binding the contract and calling it. The second method is way slower and hurts sync performance.

Reply:
perhaps this cuts corners. . .  but the real problem being avoided here is that the appealDecision handler doesn't detect subcourt jumps so reading the contract avoids depending on the graph schema. We can fix this properly though. . .

Reply:
I added a court jump handler and reverted by changes to lookup the court entity.